In [31]:
from sklearn.model_selection import cross_val_score
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.pipeline import make_pipeline
from sklearn.compose import make_column_transformer
from sklearn.metrics import accuracy_score
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC, LinearSVC
from xgboost import XGBClassifier
from sklearn.ensemble import RandomForestClassifier, AdaBoostClassifier, GradientBoostingClassifier
from sklearn.neighbors import KNeighborsClassifier

In [32]:
x_train=pd.read_csv('scaled_X_train.csv')
x_test=pd.read_csv('scaled_X_test.csv')
y_train=pd.read_csv('y_train.csv')
y_test=pd.read_csv('y_test.csv')

**fitting logistic regression**

In [ ]:
lg = LogisticRegression(max_iter=1000)

cv_scores_lg = cross_val_score(lg, x_train, y_train, cv=10)
print(f"Cross-validation scores logistic regression: {cv_scores_lg}")

mean_cv_score_lg= cv_scores_lg.mean()
std_cv_score_lg= cv_scores_lg.std()
print(f"mean_score_cv_lg:", mean_cv_score_lg)
print(f"std_score_cv_lg:" ,std_cv_score_lg)

lg.fit(x_train, y_train)
print("Logistic Regression model trained successfully.")

y_train_pred_lg = lg.predict(x_train)
train_accuracy_lg = accuracy_score(y_train, y_train_pred_lg)
print(f"Training Accuracy of logistic regression: {train_accuracy_lg:.4f}")

y_test_pred_lg = lg.predict(x_test)
test_accuracy_lg = accuracy_score(y_test, y_test_pred_lg)
print(f"Test Accuracy: {test_accuracy_lg:.4f}")

**fitting linearsvc**


In [ ]:
svc = LinearSVC()

cv_scores_svc = cross_val_score(svc, x_train, y_train, cv=10)
print(f"Cross-validation scores linear SVC: {cv_scores_svc}")

mean_cv_score_svc= cv_scores_svc.mean()
std_cv_score_svc= cv_scores_svc.std()
print(f"mean_score_cv_svc:", mean_cv_score_svc)
print(f"std_score_cv_svc:" ,std_cv_score_svc)

svc.fit(x_train, y_train)
print("Linear SVC model trained successfully.")

y_train_pred_svc = svc.predict(x_train)
train_accuracy_svc = accuracy_score(y_train, y_train_pred_svc)
print(f"Training Accuracy of Linear SVC: {train_accuracy_svc:.4f}")

y_test_pred_svc = svc.predict(x_test)
test_accuracy_svc = accuracy_score(y_test, y_test_pred_svc)
print(f"Test Accuracy of Linear SVC: {test_accuracy_svc:.4f}")

**fitting XGBClassifier**

In [ ]:
xgb = XGBClassifier(n_estimators=100, learning_rate=0.1, max_depth=6, use_label_encoder=False, eval_metric='logloss')

cv_scores_xgb = cross_val_score(xgb, x_train, y_train, cv=10)
print(f"Cross-validation scores XGB: {cv_scores_xgb}")

mean_cv_score_xgb= cv_scores_xgb.mean()
std_cv_score_xgb= cv_scores_xgb.std()
print(f"mean_score_cv_xgb:", mean_cv_score_xgb)
print(f"std_score_cv_xgb:" ,std_cv_score_xgb)

xgb.fit(x_train, y_train)
print("XGB model trained successfully.")

y_train_pred_xgb = xgb.predict(x_train)
train_accuracy_xgb = accuracy_score(y_train, y_train_pred_xgb)
print(f"Training Accuracy of XGB: {train_accuracy_xgb:.4f}")

y_test_pred_xgb = xgb.predict(x_test)
test_accuracy_xgb = accuracy_score(y_test, y_test_pred_xgb)
print(f"Test Accuracy of XGB: {test_accuracy_xgb:.4f}")

**KNN Classifier**

In [ ]:
knn = KNeighborsClassifier(n_neighbors=5)

cv_scores_knn = cross_val_score(knn, x_train, y_train, cv=10)
print(f"Cross-validation scores KNN: {cv_scores_knn}")

mean_cv_score_knn= cv_scores_knn.mean()
std_cv_score_knn= cv_scores_knn.std()
print(f"mean_score_cv_knn:", mean_cv_score_knn)
print(f"std_score_cv_knn:" ,std_cv_score_knn)

knn.fit(x_train, y_train)
print("KNN model trained successfully.")

y_train_pred_knn =knn.predict(x_train)
train_accuracy_knn = accuracy_score(y_train, y_train_pred_knn)
print(f"Training Accuracy of KNN: {train_accuracy_knn:.4f}")

y_test_pred_knn = knn.predict(x_test)
test_accuracy_knn = accuracy_score(y_test, y_test_pred_knn)
print(f"Test Accuracy of KNN: {test_accuracy_knn:.4f}")

**fitting random forest classifier**

In [37]:
rfc = RandomForestClassifier(n_estimators=100, random_state=42)

cv_scores_rfc = cross_val_score(rfc, x_train, y_train, cv=10)
print(f"Cross-validation scores Random Forest Classifier: {cv_scores_rfc}")

mean_cv_score_rfc= cv_scores_rfc.mean()
std_cv_score_rfc= cv_scores_rfc.std()
print(f"mean_score_cv_rfc:", mean_cv_score_rfc)
print(f"std_score_cv_rfc:" ,std_cv_score_rfc)

rfc.fit(x_train, y_train)
print("Random Forest Classifier model trained successfully.")

y_train_pred_rfc =rfc.predict(x_train)
train_accuracy_rfc = accuracy_score(y_train, y_train_pred_rfc)
print(f"Training Accuracy of Random Forest Classifier: {train_accuracy_rfc:.4f}")

y_test_pred_rfc= rfc.predict(x_test)
test_accuracy_rfc = accuracy_score(y_test, y_test_pred_rfc)
print(f"Test Accuracy of Random Forest Classifier: {test_accuracy_rfc:.4f}")

c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\base.py:1473: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  return fit_method(estimator, *args, **kwargs)
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\base.py:1473: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  return fit_method(estimator, *args, **kwargs)
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\base.py:1473: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  return fit_method(estimator, *args, **kwargs)
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\base.py:1473: DataConversionWarning: A column-vector y was passed when a 1d a

Cross-validation scores Random Forest Classifier: [0.9545939  0.95821786 0.95075677 0.9479855  0.95331486 0.95352803
 0.9496909  0.94734598 0.95011725 0.94776119]
mean_score_cv_rfc: 0.951331224924196
std_score_cv_rfc: 0.003333874373492677


c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\base.py:1473: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  return fit_method(estimator, *args, **kwargs)


Random Forest Classifier model trained successfully.
Training Accuracy of Random Forest Classifier: 1.0000
Test Accuracy of Random Forest Classifier: 0.9511


**fitting ADABoost**

In [38]:
adabc = AdaBoostClassifier(n_estimators=100, learning_rate=1.0, random_state=42)

cv_scores_adabc = cross_val_score(adabc, x_train, y_train, cv=10)
print(f"Cross-validation scores AdaBoostClassifier: {cv_scores_adabc}")

mean_cv_score_adabc= cv_scores_adabc.mean()
std_cv_score_adabc= cv_scores_adabc.std()
print(f"mean_score_cv_adabc:", mean_cv_score_adabc)
print(f"std_score_cv_adabc:" ,std_cv_score_adabc)

adabc.fit(x_train, y_train)
print("AdaBoostClassifier model trained successfully.")

y_train_pred_adabc =adabc.predict(x_train)
train_accuracy_adabc = accuracy_score(y_train, y_train_pred_adabc)
print(f"Training Accuracy of AdaBoostClassifier: {train_accuracy_adabc:.4f}")

y_test_pred_adabc= adabc.predict(x_test)
test_accuracy_adabc = accuracy_score(y_test, y_test_pred_adabc)
print(f"Test Accuracy of AdaBoostClassifier: {test_accuracy_adabc:.4f}")

c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\utils\validation.py:1339: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\ensemble\_weight_boosting.py:527: FutureWarning: The SAMME.R algorithm (the default) is deprecated and will be removed in 1.6. Use the SAMME algorithm to circumvent this warning.
  warnings.warn(
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\utils\validation.py:1339: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\ensemble\_weight_boosting.py:527: FutureWarning: The SAMME.R algorithm (the default) is deprecated and

Cross-validation scores AdaBoostClassifier: [0.93114475 0.93221062 0.9298657  0.92112556 0.92645491 0.93263696
 0.92538904 0.92347048 0.92517587 0.92366738]
mean_score_cv_adabc: 0.9271141263563717
std_score_cv_adabc: 0.003853303976770112


c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\utils\validation.py:1339: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\ensemble\_weight_boosting.py:527: FutureWarning: The SAMME.R algorithm (the default) is deprecated and will be removed in 1.6. Use the SAMME algorithm to circumvent this warning.
  warnings.warn(


AdaBoostClassifier model trained successfully.
Training Accuracy of AdaBoostClassifier: 0.9274
Test Accuracy of AdaBoostClassifier: 0.9239


**fitting GradientBoostingClassifier**

In [39]:
gbc = GradientBoostingClassifier(n_estimators=100, learning_rate=0.1, random_state=42)

cv_scores_gbc = cross_val_score(gbc, x_train, y_train, cv=10)
print(f"Cross-validation scores GradientBoostingClassifier: {cv_scores_gbc}")

mean_cv_score_gbc= cv_scores_gbc.mean()
std_cv_score_gbc= cv_scores_gbc.std()
print(f"mean_score_cv_gbc:", mean_cv_score_gbc)
print(f"std_score_cv_gbc:" ,std_cv_score_gbc)

gbc.fit(x_train, y_train)
print("GradientBoostingClassifier model trained successfully.")

y_train_pred_gbc =gbc.predict(x_train)
train_accuracy_gbc = accuracy_score(y_train, y_train_pred_gbc)
print(f"Training Accuracy of GradientBoostingClassifier: {train_accuracy_gbc:.4f}")

y_test_pred_gbc= gbc.predict(x_test)
test_accuracy_gbc = accuracy_score(y_test, y_test_pred_gbc)
print(f"Test Accuracy of GradientBoostingClassifier: {test_accuracy_gbc:.4f}")

c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\preprocessing\_label.py:114: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\preprocessing\_label.py:114: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\preprocessing\_label.py:114: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\preprocessing\_label.py:114: DataConversionWarning: A column-vector y was 

Cross-validation scores GradientBoostingClassifier: [0.95288851 0.95779152 0.94862503 0.94585376 0.95096994 0.95288851
 0.9496909  0.94691963 0.94713281 0.94840085]
mean_score_cv_gbc: 0.9501161458293088
std_score_cv_gbc: 0.003429937166158381


c:\Users\umert\anaconda3\envs\bigdata_env\lib\site-packages\sklearn\preprocessing\_label.py:114: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


GradientBoostingClassifier model trained successfully.
Training Accuracy of GradientBoostingClassifier: 0.9515
Test Accuracy of GradientBoostingClassifier: 0.9503


| **Model**                | **Training Accuracy** | **Test Accuracy** | **CV Scores Mean** |
|--------------------------|-----------------------|-------------------|--------------------|
| Logistic Regression      | 0.9126                | 0.9107            | 0.9125             |
| Linear SVC               | 0.9102                | 0.9082            | 0.9099             |
| Random Forest            | 1.0000                | 0.9511            | 0.9513             |
| **XGBoost**              | **0.9564**            | **0.9529**        | **0.9521**         |
| AdaBoost                 | 0.9274                | 0.9239            | 0.9271             |
| K-Nearest Neighbors      | 0.9463                | 0.9332            | 0.9323             |
| Gradient Boosting        | 0.9515                | 0.9503            | 0.9501             |